### ISSUE TITLE
Can not publish to a private NPM registry

### ISSUE THREAD

**Issue by artemtam** [release-it/release-it#700]
When publishing to a private NPM registry, I get an error that says `ERROR User<username> is not a collaborator for <@scope/package-name>`.

Adding `--npm.skipChecks` flag fixes it; however, it makes sense to check if the package is going to be published to the custom npm registry (it is set in `publishConfig`) and skip them automatically.

_Originally posted by @artemtam in https://github.com/release-it/release-it/issues/697#issuecomment-688199006_

**Comment by webpro** [comment]
Could you please do a debug run with e.g.:

```
DEBUG=release-it:* release-it [...]
```

Then we can see further, it's a command like this:

```
npm access ls-collaborators <@scope/package-name> <registry>
```

I'm especially interested in the response from the registry.

**Comment by artemtam** [comment]
The registry return `400 Bad Request` on `npm access ls-collaborators` command. 

I use Nexus private npm repository, so it turns out it is not release-it issue. I think Nexus just doesn't implement `npm access` API or I have misconfigured the permissions. Anyway, it must not be a release-it issue, even though it appeared after the update.

**Comment by webpro** [comment]
This collaborator check has just been added in v14, and unfortunately for a Nexus environment this turns out to be a breaking change.

I've heard the same issues for the `npm ping` and `npm whoami` commands, they already have an exception in release-it's codebase. So, in practice, the `--npm.skipChecks` would be a more performant solution to skip these checks altogether as release-it ignores such specific error responses (`E400` and `E404`) anyway.

However, I'm going to add the `400` response exception (the `404` is already covered for unreleased packages) for Nexus (and maybe other registry solutions). Then users will also see the corresponding warning(s).

Could you please provide me with the full response that probably includes the `E400` error code from npm? That will wil the implementation and adding a test for this.

**Comment by Alonski** [comment]
@webpro Also running into this issue with Nexus.
Here is the error that `release-it` is throwing:
```
+ release-it minor
ERROR User RELEASE_BOT is not a collaborator for @camilyo/camilyo-components.
Documentation: https://git.io/release-it-npm
```
When running `npm access ls-collaborators --registry=myregistry.com`
I get this error:
```
➜  folder git:(feature/app) npm access ls-collaborators --registry=myregistry.com
npm ERR! code E400
npm ERR! 400 Bad Request - GET myregistry.com/collaborators?format=cli

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2020-09-14T08_25_40_511Z-debug.log
```

**Comment by webpro** [comment]
Thanks @Alonski! That's what I needed. Just released a fix in [v14.0.3](https://github.com/release-it/release-it/releases/tag/14.0.3)

**Comment by Alonski** [comment]
Thanks @webpro !
